src/hotspot/share/opto/escape.cpp

```diff
@@ -3503,7 +3503,7 @@ bool ConnectionGraph::not_global_escape(Node *n) {
 // all compiled code paths have corresponding Lock/Unlock pairs.
 bool ConnectionGraph::can_eliminate_lock(AbstractLockNode* alock) {
   BoxLockNode* box = alock->box_node()->as_BoxLock();
```

Comment:
The definition can be moved under `EliminateNestedLocks` guard.

Reply:
Done.